tpm2/tpm2.go

```diff
@@ -1447,8 +1464,14 @@ func decodeSign(buf []byte) (*Signature, error) {
 
 // SignWithSession computes a signature for digest using a given loaded key. Signature
 // algorithm depends on the key type. Used for keys with non-password authorization policies.
-func SignWithSession(rw io.ReadWriter, sessionHandle, key tpmutil.Handle, password string, digest []byte, sigScheme *SigScheme) (*Signature, error) {
-	cmd, err := encodeSign(sessionHandle, key, password, digest, sigScheme)
+// If 'key' references a Restricted Decryption key, 'validation' must be a valid hash verification
+// ticket from the TPM, which can be obtained by using Hash() to hash the data with the TPM.
+// If 'validation' is nil, a NULL ticket is passed to TPM2_Sign.
+func SignWithSession(rw io.ReadWriter, sessionHandle, key tpmutil.Handle, password string, digest []byte, sigScheme *SigScheme, validation *Ticket) (*Signature, error) {
+	if validation == nil {
+		validation = &nullTicket
+	}
```

Comment:
Nit: have `encodeSign` take a `*Ticket` instead of a `Ticket`, then handle the `nil` encoding in `encodeSign`. This makes the tests easier to read, and avoids needing `nullTicket` to be declared outside of `encodeSign`.

Reply:
It's also more consistent with how we handle `SigScheme`.

Reply:
Good call, thanks!